hash.js

```diff
@@ -0,0 +1,40 @@
+const crypto = require('crypto');
+
+function hash(algorithm, message, inputEncoding, outputEncoding) {
+  const func = crypto.createHash(algorithm);
+  func.update(message, inputEncoding);
+  return func.digest(outputEncoding);
+}
+
+function hashStream(algorithm, input, inputEncoding) {
+  return new Promise((resolve, reject) => {
+    let data;
+    let isBuffer = false;
+
+    input.on('data', chunk => {
+      if (Buffer.isBuffer(chunk)) {
+        isBuffer = true;
+        chunk = chunk.toString('utf8');
```

Comment:
You need to accumulate buffers and strings independently. Calling .toString() will cause some different results for unicode strings I presume.